src/symbols/Node.tsx

```diff
@@ -16,6 +16,7 @@ import { useDrag } from '../utils/useDrag';
 export interface NodeProps {
   id: string;
   theme: Theme;
+  parent?: string;
```

Comment:
Couldn't nodes have more than 1 parent? I think you will need to use the graph object to get incoming edges.